### ISSUE TITLE
Landscape should default to HTTPS if available

### ISSUE THREAD

**Issue by sopmep-malpub** [urbit/urbit#2807]
When I access `<my-urbit-id>.arvo.network` over mobile and without specifying the protocol scheme, Safari by default loads the unencrypted HTTP endpoint. For Urbits with HTTPS configured, I think we either should be using something like HSTS or disabling the insecure endpoint.

I'm currently running `0v1e.va6cc.cks0i.8dhed.sre6p.pi67c.4b9he.d9c5s.ii3v2.c2v76.i2ffu`. Not sure if this behavior has changed in the latest revision but I haven't been able to install it yet.

**Comment by sopmep-malpub** [comment]
I just noticed another HTTPS issue.

> Log into Landscape using HTTPS.
> Navigate `Chats > ~/~dabben-larbet/urbit-weekend`
> The protocol has now downgraded from HTTPS to HTTP

I don't experience this behavior clicking on any particular chat, though. For example, I don't experience a downgrade if I navigate `Chats > ~hiddev-dannut/hooniverse-chat`.

This was on Chrome 81.0.4044.122 on Mac.

Let me know if I should create another thread for this.

**Comment by Fang-** [comment]
Looks like `+http-config` already contains a flag for doing this?

https://github.com/urbit/urbit/blob/3196179fbdc7afe4f5d4c8143b85ee7c5b9c7d41/pkg/arvo/sys/zuse.hoon#L935-L939

But `http.c`'s server initialization doesn't seem to respect it yet.

https://github.com/urbit/urbit/blob/3196179fbdc7afe4f5d4c8143b85ee7c5b9c7d41/pkg/urbit/vere/http.c#L1170-L1176

I don't know too much about h2o or our use of it, so can't say if filling out that `XX` is trivial or not.

**Comment by joemfb** [comment]
It's pretty trivial. I had it on a branch at some point, but it'd be easy to rewrite. NB, the frontend should also scheme-agnostic `//foo.bar/baz/bud` urls where applicable.

**Comment by Fang-** [comment]
So I tried doing this. Even with @joemfb's help, it turned out to be decidedly non-trivial. h2o expects you to supply the new "prefix" to forward to. This prefix must include the domain name and optional port. The way `http.c` is currently set up, configuring the server to set up redirects like that would require a partial rewrite.

You could still do this on the eyre side, of course, which already reads the `Host` header for the target domain. But regardless of where you try to solve this, there's another problem you run into: you have no way of knowing where/how you're actually running or reachable.

If urbit opened the https endpoint on a non-standard port, you'll want to include that in the redirect. This works fine if you're exposed to the internet directly, but not if you're behind a reverse proxy. It may listen on the standard port, or something different altogether. 

In other words, we can attempt a redirect, but there's no way to guarantee the target will actually work. With that, I'm not sure this redirect should be done by urbit itself in the first place...

**Comment by galenwp** [comment]
Can Urbit know that you're running over raw HTTP and warn you / link to instructions? That alone would be sufficient. Browsers often let you go to `http://` when `https://` is available.

**Comment by vvisigoth** [comment]
btw the automated hosting controller redirects by default.
—
~poldec-tonteg
http://urbit.org
